lib/cloudware/cli.rb

```diff
@@ -277,6 +277,80 @@ class CLI
+        c.syntax = 'cloudware machine power status [options]'
+        c.description = 'Check the power status of a machine'
+        c.option '--name NAME', String, 'Machine name'
+        c.option '--domain NAME', String, 'Domain identifier'
+        c.action do |_args, options|
+          machine = Cloudware::Machine.new
+          options.name = ask('Machine name: ') if options.name.nil?
+          machine.name = options.name.to_s
+
+          options.domain = ask('Domain identifier: ') if options.domain.nil?
+          machine.domain = options.domain.to_s
+
+          puts "#{options.name}: Power status is #{machine.get_item('state')}"
+        end
+      end
+
+      command :'machine power on' do |c|
+        c.syntax = 'cloudware machine power on [options]'
+        c.description = 'Turn a machine on'
+        c.option '--name NAME', String, 'Machine name'
+        c.option '--domain NAME', String, 'Domain identifier'
+        c.action do |_args, options|
+          machine = Cloudware::Machine.new
+          options.name = ask('Machine name: ') if options.name.nil?
+          machine.name = options.name.to_s
+
+          options.domain = ask('Domain identifier: ') if options.domain.nil?
+          machine.domain = options.domain.to_s
+
+          Whirly.start spinner: 'dots2', status: "Powering on machine #{options.name}".bold, stop: '[OK]'.green
+          machine.power_on
+          Whirly.stop
+        end
+      end
+
+      command :'machine power off' do |c|
+        c.syntax = 'cloudware machine power off [options]'
+        c.description = 'Turn a machine off'
+        c.option '--name NAME', String, 'Machine name'
+        c.option '--domain NAME', String, 'Domain identifier'
+        c.action do |_args, options|
+          machine = Cloudware::Machine.new
+          options.name = ask('Machine name: ') if options.name.nil?
+          machine.name = options.name.to_s
+
+          options.domain = ask('Domain identifier: ') if options.domain.nil?
+          machine.domain = options.domain.to_s
+
+          Whirly.start spinner: 'dots2', status: "Powering off machine #{options.name}".bold, stop: '[OK]'.green
+          machine.power_off
+          Whirly.stop
+        end
+      end
+
+      command :'machine rebuild' do |c|
+        c.syntax = 'cloudware machine rebuild [options]'
+        c.description = 'Rebuild a machine'
+        c.option '--name NAME', String, 'Machine name'
+        c.option '--domain NAME', String, 'Domain identifier'
+        c.action do |_args, options|
+          machine = Cloudware::Machine.new
+          options.name = ask('Machine name: ') if options.name.nil?
+          machine.name = options.name.to_s
+
+          options.domain = ask('Domain identifier: ') if options.domain.nil?
+          machine.domain = options.domain.to_s
+
+          Whirly.start spinner: 'dots2', status: "Recreating machine #{options.name}".bold, stop: '[OK]'.green
+          machine.rebuild
+          Whirly.stop
```

Comment:
You could simplify this using the block syntax for `Whirly.start` (see https://github.com/janlelis/whirly#basic-usage).

Also since in a lot of places we are calling `Whirly.start` with the same arguments, apart from `status`, you could extract a method for doing this, something like:

```ruby
def with_spinner(status)
  Whirly.start do spinner: 'dots2', status: status.bold, stop: '[OK]'.green
    yield
  end
end
```

after which I believe the above would just become:

```ruby
  with_spinner "Recreating machine #{options.name}" do
    machine.rebuild
  end
```